### PR TITLE
Bug 1918788: Avoid unneeded object initialization during profile conversion 

### PR DESCRIPTION
**Pull request by cynepco3hahue**
It possible the situation when the source conversion object has nil for
some field, but we initialize this field under the destination object.

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

File: api/v1alpha1/performanceprofile_conversion.go [hunk 1]
```diff
@@ -17,8 +17,9 @@ func (curr *PerformanceProfile) ConvertTo(dstRaw conversion.Hub) error {
 	dst.ObjectMeta = curr.ObjectMeta
 
 	// Spec
-	dst.Spec.CPU = new(v1.CPU)
 	if curr.Spec.CPU != nil {
+		dst.Spec.CPU = new(v1.CPU)
```
Comment (fromanirh) [inline api/v1alpha1/performanceprofile_conversion.go:21]: I don't follow. We know `curr.Spec.CPU` is non-nil so... we initialize it again? (here and in a bunch of other places)

**Review by fromanirh — COMMENTED** [review]
I don't really understand why the logic is the reverse of what I'd expect

File: api/v1alpha1/performanceprofile_conversion.go (hunk 1 above)
Comment (cynepco3hahue) [reply api/v1alpha1/performanceprofile_conversion.go:21]: we check the `curr` one, and if not nil we initialize the destination one.

Comment (fromanirh) [reply api/v1alpha1/performanceprofile_conversion.go:21]: ah right, stupid me. Reading again.

File: functests/1_performance/performance.go [hunk 2]
```diff
@@ -474,41 +507,49 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
...
-			v1Profile.ResourceVersion = ""
-			v1Profile.Spec.NodeSelector = map[string]string{"test": "test"}
-			Expect(testclient.Client.Create(context.TODO(), v1Profile)).ToNot(HaveOccurred())
+		}
 
-			key = types.NamespacedName{
-				Name:      v1Profile.Name,
-				Namespace: v1Profile.Namespace,
-			}
+		When("the performance profile does not NUMA field", func() {
```
Comment (fromanirh) [inline functests/1_performance/performance.go:512]: does not contain?

**Review by fromanirh — COMMENTED** [review]
just a minor comment about a comment, and looks good.

File: api/v1alpha1/performanceprofile_conversion.go (hunk 1 above)
Comment (cynepco3hahue) [reply api/v1alpha1/performanceprofile_conversion.go:21]: end of the week :)

File: functests/1_performance/performance.go (hunk 2 above)
Comment (cynepco3hahue) [reply functests/1_performance/performance.go:512]: done

**Review by fromanirh — COMMENTED** [review]
/lgtm
/hold
to give other reviewers time

**Comment by Avielyo10** [comment]
How didn't we catch it sooner?

**Comment by cynepco3hahue** [comment]
Because we used the profile with the NUMA field.

**Review by MarSik — APPROVED** [review]

**Pull request merged**